### PR TITLE
Added Customizable Name Field from User Mapping

### PR DESCRIPTION
**Pull request by ericbram**
Added a UI configuration that allows users to replace the xAPI
Actor->Account->Name field with a mapping other than the user's Moodle
Id.

### CODE REVIEW

**Comment by ryasmi** [comment]
Hey @ericbram, unfortunately we can't merge this until the Travis build passes, [it's failing right now](https://travis-ci.org/xAPI-vle/moodle-logstore_xapi/builds/253717162?utm_source=github_status&utm_medium=notification).

```
PHP Fatal error:  Undefined class constant 'AUTHSYNCFIELDS' in /home/travis/build/moodle/admin/tool/log/store/xapi/settings.php on line 33
```

File: lib/translator/src/Events/Event.php [hunk 1]
```diff
@@ -31,12 +31,14 @@ class Event extends PhpObj {
      */
     public function read(array $opts) {
         $appname = $opts['app']->fullname ?: 'A Moodle site';
+		$subField = get_config('logstore_xapi', 'account') ?: 'id';
```
Comment (ryasmi) [inline lib/translator/src/Events/Event.php:34]: For testing purposes, the `get_config` function should not be used in the Translator. Pass this through the constructor instead.

File: settings.php [hunk 2]
```diff
@@ -29,6 +29,9 @@
 use \MXTranslator\Controller as translator_controller;
 
 if ($hassiteconfig) {
+	// get list of user map settings for account association
+	$accountOpts = array_merge(['id'], \core_user::AUTHSYNCFIELDS);
```
Comment (ryasmi) [inline settings.php:33]: Assuming something is wrong here since attempting to use `AUTHSYNCFIELDS` throws an error.

**Review by ryasmi — CHANGES_REQUESTED** [review]

**Comment by ericbram** [comment]
Thanks for taking a look at this, I am not familiar with contributing to other projects yet and actually thought I was still merging to my forked version!  I'll have to take a look at the AUTHSYNCFIELDS error when I get a chance, that code seemed to have been working fine for me locally.

**Comment by ryasmi** [comment]
You're welcome @ericbram 👍 Haha no worries. Okay thank you, yeah that doesn't surprise me, it'll just be because Moodle isn't technically running during that build. We have to keep our tests quite isolated.

**Comment by ericbram** [comment]
Understood!  I was trying to re-use that list of user mapping options and that seemed to be the only array list.  I'll see if I can find a way to make that accessible or if I have to re-create that list.

**Comment by ryasmi** [comment]
Okay nice :smile:

**Comment by bseclier** [comment]
Hi,
Any news about this PR ? We are currently looking for this feature too (we'll patch our code locally by now, but it would be nice adding this feature in the official source code).
Thanks,

**Comment by ericbram** [comment]
Hi,

Unfortunately I had to move on as I was doing that at the request of my
company and they have since shifted priorities.  Feel free to use any of
the code I had written to complete the PR as necessary.  If I remember
right, I was just trying to avoid re-implementing an ENUM which was causing
the issue.


Thanks,
Eric Bram

**Comment by ryasmi** [comment]
Closing due to inactivity.

**Pull request closed without merging**